UPGRADING.md

```diff
@@ -1,218 +1,144 @@
+Disable automatic restart of the node service. To do so please alter your `gaiad.service` file configuration and set appropriate lines to following values.
 
-```shell
-ERR UPGRADE "v15" NEEDED at height: <UPGRADE_HEIGHT>: upgrade to v15 and applying upgrade "v15" at height:<UPGRADE_HEIGHT>
 ```
+Restart=no 
 
-Stop the node, and switch the binary to **Gaia v15.1.0** and re-start by `gaiad start`.
-
-It may take several minutes to a few hours until validators with a total sum voting power > 2/3 to complete their node upgrades. After that, the chain can continue to produce blocks.
+Environment="DAEMON_ALLOW_DOWNLOAD_BINARIES=false"
+Environment="DAEMON_RESTART_AFTER_UPGRADE=false"
+```
 
-### Method II: Upgrade using Cosmovisor
+After that you will need to run `sudo systemctl daemon-reload` to apply changes in the service configuration.
 
-#### Manually preparing the binary
+There is no need to restart the node yet; these changes will get applied during the node restart in the next step.
 
-##### Preparation
+# Setup Cosmovisor
+## Create the updated gaiad binary of v15.2.0
 
-- Install the latest version of Cosmovisor (`1.5.0`):
+### Go to gaiad directory if present else clone the repository
 
 ```shell
-go install cosmossdk.io/tools/cosmovisor/cmd/cosmovisor@latest
-cosmovisor version
-# cosmovisor version: v1.5.0
+   git clone https://github.com/cosmos/gaia.git
 ```
 
-- Create a `cosmovisor` folder inside `$GAIA_HOME` and move Gaia `v14.2.0` into `$GAIA_HOME/cosmovisor/genesis/bin`:
+### Follow these steps if gaiad repo already present
 
 ```shell
-mkdir -p $GAIA_HOME/cosmovisor/genesis/bin
-cp $(which gaiad) $GAIA_HOME/cosmovisor/genesis/bin
+   cd $HOME/.gaia
+   git pull
+   git fetch --tags
+   git checkout v15.2.0
+   make install
 ```
 
-- Build Gaia `v15.1.0`, and move gaiad `v15.1.0` to `$GAIA_HOME/cosmovisor/upgrades/v15/bin`
-
+### Check the new gaiad version, verify the latest commit hash
 ```shell
-mkdir -p  $GAIA_HOME/cosmovisor/upgrades/v15/bin
-cp $(which gaiad) $GAIA_HOME/cosmovisor/upgrades/v15/bin
+   $ gaiad version --long
+   name: gaiad
+   server_name: gaiad
+   version: 14.2.0
+   commit: <commit-hash>
+   ...
 ```
 
-At this moment, you should have the following structure:
+### Or check checksum of the binary if you decided to download it
 
 ```shell
-.
-├── current -> genesis or upgrades/<name>
-├── genesis
-│   └── bin
-│       └── gaiad  # old: v14.2.0
-└── upgrades
-    └── v15
-        └── bin
-            └── gaiad  # new: v15.1.0
+$ shasum -a 256 gaiad-v15.2.0-linux-amd64
+<checksum>  gaiad-v15.2.0-linux-amd64
 ```
 
-- Export the environmental variables:
-
+## Copy the new gaiad (v15.2.0) binary to cosmovisor current directory
 ```shell
-export DAEMON_NAME=gaiad
-# please change to your own gaia home dir
-# please note `DAEMON_HOME` has to be absolute path
-export DAEMON_HOME=$GAIA_HOME
-export DAEMON_RESTART_AFTER_UPGRADE=true
+   cp $GOPATH/bin/gaiad ~/.gaiad/cosmovisor/current/bin
```

Comment:
not sure but isn't there a directory structure which contains old and new version of gaiad  and current is just a link pointing to one of them. don't we need to create a directory for the new version where the new version needs to be copied instead of current?

Reply:
This is not a coordinated upgrade, cosmovisor doesn't have any upgrade info so you need to do it manually.

It's a "hot swap" so to speak.